### ISSUE TITLE
Feature Request: Easier clearing of remapped buttons.

### ISSUE THREAD

**Issue by M-Walrus** [MiSTer-devel/Main_MiSTer#781]
Currently you need a keyboard connected to clear buttons that have been remapped by using the ESC key.  I propose that another option be added so that folks who are using MiSTer with only controllers connected or in an arcade cabinet can easily undo the remappings as well, without relying on a keyboard.

Method:  Hold the Menu/OSD button for 3 seconds and the controller has all mappings reset, while in the button remapping menu.  This should only effect the device that is currently being used, and not other controllers that are connected and have had their buttons remapped.  A similar hold-OSD function is already used in the controller binding menu to cancel binding a controller and seems like it would work well here.

**Comment by sorgelig** [comment]
i will check this possibility

**Comment by sorgelig** [comment]
It's already implemented using USER button. Press any button on joystick first, then hold USER button and mapping of this joystick to specific core will be erased, so default mapping will be used.

**Comment by M-Walrus** [comment]
Ah I didn't realize that.  I tend to try and not hit that button when a core is loaded because it will reset generally.  I've accidentally pressed it when trying to hit the OSD key before so tend to try and not use it.  If possible to do with the actual bound OSD/menu key on a controller it would be ideal.  If it's too much work to implement, no biggie.

**Comment by sorgelig** [comment]
Controller's OSD/Menu button is already used for "undefine" with short press and "cancel" with long press.

**Comment by M-Walrus** [comment]
Yes, in the core controller binding screen that is how it functions.  In the remapping screen I would like to see the long press of OSD/menu button on a controller able to revert all bindings to their original locations.  Currently long press of OSD/menu button on a controller doesn't seem to do anything in the rebinding screen, which is why it seemed a good way to add this functionality.

**Comment by sorgelig** [comment]
by OSD/menu - i mean the button on controller, not on I/O board.

**Comment by sixtyfpsokthx** [comment]
I ran into [this problem](https://discord.com/channels/647909397477195803/737505468670279710/1159368553015803974) also.  It's disappointing that undoing the act of "temporarily" remapping buttons is still so difficult.  I had to delete a file on my SD card to get the normal mapping back, after trying every less intrusive approach I could think of.

**Comment by rsn8887** [comment]
> I ran into [this problem](https://discord.com/channels/647909397477195803/737505468670279710/1159368553015803974) also. It's disappointing that undoing the act of "temporarily" remapping buttons is still so difficult. I had to delete a file on my SD card to get the normal mapping back, after trying every less intrusive approach I could think of.

It works, but you have to press some button on the controller you want to clear first, so MiSTer knows what to clear. See here: https://github.com/MiSTer-devel/Main_MiSTer/issues/898#issuecomment-2332950458

**Comment by rsn8887** [comment]
See here: https://github.com/MiSTer-devel/Main_MiSTer/issues/898#issuecomment-2332950458
